app/transformers/PAYETransformer.scala

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package transformers
+
+import models.Amount
+import play.api.Mode.Mode
+import play.api.{Configuration, Play}
+import play.api.libs.json._
+import play.api.libs.json.JsPath
+import uk.gov.hmrc.play.config.ServicesConfig
+
+trait PAYETransformer {
```

Comment:
This is a really complex transformer, is it only this complex because you're using safeTransform to always return a JsObject where really you want to fail and have a None[JsObject] rather than return the original, which is presumably unusable in the frontend?

Reply:
I think we should merge and raise this as a separate ticket. I've been looking at the error handling for SA. I don't understand it but we may want to do something similar.